server/routes/projects.js

```diff
@@ -0,0 +1,55 @@
+const { ObjectId } = require('bson');
+const express = require('express');
+const projectsData = require('../data/projects');
+const {
+	sendErrResp,
+	isValidArray,
+	isValidObj,
+	isValidStr,
+} = require('../utils');
+const { isValidProjectName } = require('../utils/projects');
+const router = express.Router();
+const { authenticateToken } = require('../middleware/auth');
+
+router.route('/project').post(authenticateToken, async (req, res) => {
+	const { user } = req;
+	const {
+		name,
+		description,
+		github,
+		media,
+		technologies,
+		owner,
+		comments,
+		likes,
+	} = req.body;
+	try {
+		// name,description,github,media,technologies,owner,comments,likes
+		// make the validations sent as parameters accurate later
+		name = isValidProjectName(name);
+		description = isValidStr(description);
+		github = isValidStr(github);
+		media = isValidArray(media);
+		technologies = isValidArray(technologies);
+		owner = isValidObj(owner);
```

Comment:
you need to validate the keys in the user object, this validation will pass for empty objects. we want to validate the ObjectId and username here

Also, need to make sure the user exists in the DB

Reply:
> bro, test your code before creating PR. you are re-assigning values to const variables in the projects routes file. And try to reuse functions instead of validating each key separately even in the routes file (this doesn't affect functionality, just a suggestion)

Sorry bro, I will look into the things which you have commented and will submit again.

Reply:
> you need to validate the keys in the user object, this validation will pass for empty objects. we want to validate the ObjectId and username here
> 
> Also, need to make sure the user exists in the DB

Bro didn't get this comment.. what do you mean by user object ??